pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -336,18 +336,21 @@ func (plugin *kubenetNetworkPlugin) setup(namespace string, name string, id kube
 
 	// Explicitly assign mac address to cbr0. If bridge mac address is not explicitly set will adopt the lowest MAC address of the attached veths.
 	// TODO: Remove this once upstream cni bridge plugin handles this
-	link, err := netlink.LinkByName(BridgeName)
-	if err != nil {
-		return fmt.Errorf("failed to lookup %q: %v", BridgeName, err)
-	}
-	macAddr, err := generateHardwareAddr(plugin.gateway)
-	if err != nil {
-		return err
-	}
-	glog.V(3).Infof("Configure %q mac address to %v", BridgeName, macAddr)
-	err = netlink.LinkSetHardwareAddr(link, macAddr)
-	if err != nil {
-		return fmt.Errorf("Failed to configure %q mac address to %q: %v", BridgeName, macAddr, err)
+	var macAddr net.HardwareAddr
+	if plugin.gateway.To4() != nil {
```

Comment:
Don't we still need this for IPv6?  Why not just generate a random MAC (setting the "locally administered" bit of course) and assign that in all cases.